payments-core/src/test/java/com/stripe/android/payments/core/authentication/GenericAuthenticatorTest.kt

```diff
@@ -0,0 +1,72 @@
+package com.stripe.android.payments.core.authentication
+
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.testing.TestLifecycleOwner
+import com.google.common.truth.Truth.assertThat
+import com.stripe.android.core.networking.ApiRequest
+import com.stripe.android.model.StripeIntent
+import com.stripe.android.view.AuthActivityStarterHost
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runTest
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.mock
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+@OptIn(ExperimentalCoroutinesApi::class)
+class GenericAuthenticatorTest {
```

Comment:
Using this to test the lifecycle observation behavior. The tests for the authenticator subclasses stay unchanged (other than renaming `authenticate` to `performAuthentication`).